data.tf

```diff
@@ -5,6 +5,7 @@ locals {
   worker_ami_name_filter_windows = (var.worker_ami_name_filter_windows != "" ?
     var.worker_ami_name_filter_windows : "Windows_Server-2019-English-Core-EKS_Optimized-${tonumber(var.cluster_version) >= 1.14 ? var.cluster_version : 1.14}-*"
   )
+  ec2_principal = (data.aws_partition.current.partition == "aws-cn") ? "ec2.amazonaws.com.cn" : "ec2.amazonaws.com"
```

Comment:
`data.aws_partition` has a `dns_suffix` output. You can use it to simplify this logic.

Reply:
Done in new commit, together with fix in aws-auth for correct arn partition